chipflow_lib/pin_lock.py

```diff
@@ -0,0 +1,177 @@
+import inspect
+import logging
+
+from pprint import pformat
+from pathlib import Path
+
+from chipflow_lib import _parse_config
+from chipflow_lib.platforms import PACKAGE_DEFINITIONS, PIN_ANNOTATION_SCHEMA, top_interfaces
+from chipflow_lib.platforms.utils import LockFile, Package, PortMap
+
+# logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)
+logger = logging.getLogger(__name__)
+
+
+def count_pins(name, member):
+    logger.debug(f"count_pins {name} {member['type']} {member['annotations'] if 'annotations' in member else 'no annotations'}")
+    if member['type']=='interface' and 'annotations' in member \
+       and PIN_ANNOTATION_SCHEMA in member['annotations']:
+        return member['annotations'][PIN_ANNOTATION_SCHEMA]['width']
+    elif member['type'] == 'interface':
+        width = 0
+        for n, v in member['members'].items():
+            width += count_pins('_'.join([name,n]), v)
+        return width
+    elif member['type'] == 'port':
+        return member['width']
+
+def count_old_pins(d):
```

Comment:
can we have some explanation here of `count_old_pins` vs `count_pins`?

Reply:
count_old_pins now removed!